packages/languages/src/node/language-server-contribution.ts

```diff
@@ -76,23 +75,54 @@ export abstract class BaseLanguageServerContribution implements LanguageServerCo
     protected async createProcessSocketConnection(outSocket: MaybePromise<net.Socket>, inSocket: MaybePromise<net.Socket>,
         command: string, args?: string[], options?: cp.SpawnOptions): Promise<IConnection> {
 
-        const process = this.spawnProcess(command, args, options);
+        const process = await this.spawnProcessAsync(command, args, options);
         const [outSock, inSock] = await Promise.all([outSocket, inSocket]);
         return createProcessSocketConnection(process.process, outSock, inSock);
     }
 
+    /**
+     * @deprecated use `createProcessStreamConnectionAsync` instead.
+     * Otherwise, the backend cannot notify the client if the LS has failed at start-up.
+     */
     protected createProcessStreamConnection(command: string, args?: string[], options?: cp.SpawnOptions): IConnection {
         const process = this.spawnProcess(command, args, options);
         return createStreamConnection(process.output, process.input, () => process.kill());
     }
 
+    protected async createProcessStreamConnectionAsync(command: string, args?: string[], options?: cp.SpawnOptions): Promise<IConnection> {
+        const process = await this.spawnProcessAsync(command, args, options);
+        return createStreamConnection(process.output, process.input, () => process.kill());
+    }
+
+    /**
+     * @deprecated use `spawnProcessAsync` instead.
+     */
     protected spawnProcess(command: string, args?: string[], options?: cp.SpawnOptions): RawProcess {
         const rawProcess = this.processFactory({ command, args, options });
         rawProcess.process.once('error', this.onDidFailSpawnProcess.bind(this));
         rawProcess.process.stderr.on('data', this.logError.bind(this));
         return rawProcess;
     }
 
+    protected spawnProcessAsync(command: string, args?: string[], options?: cp.SpawnOptions): Promise<RawProcess> {
+        const rawProcess = this.processFactory({ command, args, options });
+        rawProcess.process.stderr.on('data', this.logError.bind(this));
+        return new Promise<RawProcess>((resolve, reject) => {
+            rawProcess.onError(error => {
+                this.onDidFailSpawnProcess(error);
+                if (isErrnoException(error) && error.code === 'ENOENT') {
+                    const guess = command.split('\S').shift();
```

Comment:
What is this line meant to do?  It looks like you wanted to do a regex, but it isn't a RegExp object, so I think it doesn't do anything for now.

Reply:
extract the command.

Reply:
And does it work?

Reply:
I guess no, it should be `split(/\s/).shift()`.